scan-framework/src/main/java/com/getbouncer/scan/framework/ml/TensorFlowLiteAnalyzer.kt

```diff
@@ -19,33 +21,41 @@ abstract class TensorFlowLiteAnalyzer<Input, MLInput, Output, MLOutput>(
     private val debug: Boolean = false
 ) : Analyzer<Input, Unit, Output> {
 
-    protected abstract fun buildEmptyMLOutput(): MLOutput
+    protected abstract suspend fun buildEmptyMLOutput(): MLOutput
 
-    protected abstract fun interpretMLOutput(data: Input, mlOutput: MLOutput): Output
+    protected abstract suspend fun interpretMLOutput(data: Input, mlOutput: MLOutput): Output
 
-    protected abstract fun transformData(data: Input): MLInput
+    protected abstract suspend fun transformData(data: Input): MLInput
 
-    protected abstract fun executeInference(tfInterpreter: Interpreter, data: MLInput, mlOutput: MLOutput)
+    protected abstract suspend fun executeInference(tfInterpreter: Interpreter, data: MLInput, mlOutput: MLOutput)
 
     private val loggingTimer by lazy {
         Timer.newInstance(Config.logTag, "$name ${this::class.java.simpleName}", enabled = debug)
     }
 
     override suspend fun analyze(data: Input, state: Unit): Output {
         val mlInput = loggingTimer.measure("transform") {
-            transformData(data)
+            runBlocking {
```

Comment:
you should not need this call. This is only required if you're calling a suspending thread from a blocking method. Since both `analyze` and `transformData` are suspend calls, this is unnecessary.

Reply:
The problem seems to be with the logging timer measure function, not supporting suspend calls, looking for a workaround.